sirepo/job_driver/sbatch.py

```diff
@@ -87,33 +87,32 @@ def init_class(cls, job_supervisor_module):
 
     async def prepare_send(self, op):
         m = op.msg
-        try:
-            self._creds = m.pkdel('sbatchCredentials')
-            if self._srdb_root is None:
-                if not self._creds or 'username' not in self._creds:
-                    self._raise_sbatch_login_srexception('no-creds', m)
-                self._srdb_root = self.cfg.srdb_root.format(
-                    sbatch_user=self._creds.username,
-                )
-            m.userDir = '/'.join(
-                (
-                    str(self._srdb_root),
-                    sirepo.simulation_db.USER_ROOT_DIR,
-                    m.uid,
-                )
+        c = m.pkdel('sbatchCredentials')
+        if self._srdb_root is None or c:
+            if c:
+                self._creds = c
+            if not self.get('_creds') or 'username' not in self._creds:
+                self._raise_sbatch_login_srexception('no-creds', m)
+            self._srdb_root = self.cfg.srdb_root.format(
```

Comment:
Again this would only happen if someone hacked our api but if they put sbatchCredentials on a message with a different username than the agent is currently running under. Then the self._srdb_root will change (and thus the userDir and runDir will change) but the agent is still running as the previous user. There will probably be permission denied problems when trying to read/write files for the new user. If self._srdb_root is True we probably shouldn't change it.

Reply:
Correct. Hacking APIs is not important as long as it doesn't lead to a breach. It will lead to random server errors, which is fine.